augur/validate.py

```diff
@@ -100,21 +100,53 @@ def validate_json(jsonToValidate, schema, filename):
     # the user isn't stuck playing whack-a-mole.
     errors = list(schema.iter_errors(jsonToValidate))
 
+    def custom_message(error):
+        """Convert technical JSON schema errors to human-readable messages."""
+        validator = error.validator
+        short_value = shorten_as_json(error.instance, 50, "…")
+
+        if validator == "oneOf":
+            return f"{short_value} did not match one of the acceptable options below."
+        if validator == "anyOf":
+            return f"{short_value} did not match any of the acceptable options below."
+
+        if validator == "required":
+            missing_property = str(error.args[0]).split("'")[1]
+            return f"Missing required property '{missing_property}'"
+
+        if validator == "additionalProperties":
+            additional_property = list(error.instance.keys() - error.schema.get('properties', {}).keys())[0]
+            return f"Unexpected property '{additional_property}'"
+
+        if validator == "type":
+            expected_type = error.validator_value
+            actual_type = type(error.instance).__name__
+            return f"Expected {expected_type} but found {actual_type} {short_value!r}"
+
+        if validator == "const":
+            expected_value = error.validator_value
+            return f"Expected '{expected_value}' but found {error.instance!r}"
+
+        if validator == "pattern":
+            pattern = error.validator_value
+            if "[0-9X]{4}-[0-9X]{2}-[0-9X]{2}" in pattern:
+                return f"Expected a date in the format YYYY-MM-DD but found {short_value}"
```

Comment:
_non-blocking_

I'm slightly worried that the pattern `[0-9X]{4}-[0-9X]{2}-[0-9X]{2}` is not always a date...but I honestly can't think of anything else here 😅 

I do think it'd be better to only change the error message for the full pattern match in case it's other parts of the pattern that failed.

```suggestion
            if pattern == "^[0-9X]{4}-[0-9X]{2}-[0-9X]{2}$":
                return f"Expected a date in the format YYYY-MM-DD but found {short_value}"
```